scripts/tests/test_suites.sh

```diff
@@ -54,8 +54,38 @@ for j in "${iter_array[@]}"; do
         echo "  ===== Running test: $i"
         echo "          * Starting cluster server"
         rm -rf /tmp/chip_tool_config.ini
-        out/debug/chip-all-clusters-app &
-        background_pid=$!
+        # This part is a little complicated.  We want to
+        # 1) Start chip-all-clusters-app in the background
+        # 2) Pipe its output through tee so we can wait until it's ready for a
+        #    PASE handshake.
+        # 3) Save its pid off so we can kill it.
+        #
+        # The subshell with echoing of $! to a file descriptor and
+        # then reading things out of there accomplishes item 3;
+        # otherwise $! would be the last-started command which would
+        # be the tee.  This part comes from https://stackoverflow.com/a/3786955
+        # and better ideas are welcome.
+        #
+        # The stdbuf -o0 is to make sure our output is flushed through
+        # tee expeditiously; otherwise it will buffer things up and we
+        # will never see the string we want.
+
+        # Clear out our temp files so we don't accidentally do a stale
+        # read from them before we write to them.
+        rm -rf /tmp/all-clusters-log
+        rm -rf /tmp/pid
```

Comment:
Should this do a 'sync' or a 'flush' to ensure the deletion actually happened?

Reply:
This does not seem relevant in that we don't care what the on-disk state is, just what happens when we open the filename, and that part should be all set once `rm` returns.  In particular, I was running into issues where we were seeing a "Server Listening" still in the all-clusters-log from a previous iteration once I moved reading that to before reading /tmp/pid.....  But if we rm that should not happen, as far as I know.